velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -472,6 +472,37 @@ struct MillisecondFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct LastDayOfMonthFunction : public InitSessionTimezone<T>,
+                                public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE Date getLastDayOfMonth(const std::tm& time) {
+    return util::daysSinceEpochFromDate(
+        1900 + time.tm_year,
+        time.tm_mon + 1,
+        util::kLastDayOfMonth[time.tm_mon] +
+            (time.tm_mon == 1 && util::isLeapYear(1900 + time.tm_year)));
```

Comment:
Perhaps, add a comment to explain this logic. I guess we add one more day for February of leap years.